conans/client/tools/env.py

```diff
@@ -111,3 +113,53 @@ def remove_from_path(command):
 
     with environment_append({"PATH": curpath}):
         yield
+
+
+def env_diff(cmd, only_diff):
```

Comment:
This doesn't properly handle environment variables with trailing whitespace or newlines in their values.

Reply:
yeah, I am not using json serialization as you did, I believe it should help to handle such situations